// eafc-clubs-api/lib/schemas.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCHEMAS = void 0;
const yup_1 = require("yup");
const platform_1 = require("./platform");
const MATCH_TYPES = ['leagueMatch', 'playoffMatch'];
const SCHEMAS = {
    CLUB_SEARCH: (0, yup_1.object)({
        clubName: (0, yup_1.string)().required().min(1).max(32),
        platform: (0, yup_1.string)().required().oneOf(platform_1.PLATFORMS),
    }),
    OVERALL_STATS: (0, yup_1.object)({
        clubIds: (0, yup_1.string)().required(),
        platform: (0, yup_1.string)().required().oneOf(platform_1.PLATFORMS),
    }),
    MEMBER_CAREER_STATS: (0, yup_1.object)({
        clubId: (0, yup_1.string)().required(),
        platform: (0, yup_1.string)().required().oneOf(platform_1.PLATFORMS),
    }),
    MEMBER_STATS: (0, yup_1.object)({
        clubId: (0, yup_1.string)().required(),
        platform: (0, yup_1.string)().required().oneOf(platform_1.PLATFORMS),
    }),
    MATCHES_STATS: (0, yup_1.object)({
        clubIds: (0, yup_1.string)().required(),
        platform: (0, yup_1.string)().required().oneOf(platform_1.PLATFORMS),
        matchType: (0, yup_1.string)().required().oneOf(MATCH_TYPES),
    }),
    CLUB_INFO: (0, yup_1.object)({
        clubIds: (0, yup_1.string)().required(),
        platform: (0, yup_1.string)().required().oneOf(platform_1.PLATFORMS),
    }),
};
exports.SCHEMAS = SCHEMAS;
